Visibility.md

```diff
@@ -84,4 +84,85 @@ If there is a file named `lib.wgsl` in the `package_root` directory,
+fresh in one programmer’s mind, then a public/private distinction doesn’t matter
+much. On the other hand, if there are multiple engineers involved, api
+boundaries help clarify communication. There are two boundaries to consider: the
+api to code in another package and the api to code in the same package.
+
+#### Private by Default for Libraries
+
+Communication between programmers working on different packages is likely quite
+limited, so I think there’s a strong argument that strong barriers are the right
+default choice. So private by default makes sense between packages. By requiring
+annotations to make something public, we require the publisher to think about
+their api. Good fences make good neighbors and all that.
+The default choice doesn't lead to an accidental maintenance problem.
+Let’s annotate with
+e.g. `public` (or `pub` or `@public`) so that private can be the default, and package
+publishers are pushed to define an api.
+
+#### Private by Default Inside the Package
+
+Communication between programmers working on the same package might be smoother
+than communications between teams working on different packages. At the least,
+people working on the same package are probably working in the same git
+repository. Should the language help distinguish public vs private for imports
+within the package? I’d argue yes, and for the same reasons as for library publishers. A
+significant enough internal module wants a boundary to aid in ongoing
+maintenance. So I think the language should support distinguishing internal from
+api for within the package.
+
+Should the default for within package visibility be public or private? For an
+insignificant internal module where everything ought to be public, it’d be
+overhead to have to annotate everything if private is the default. But for a
+module that does want to distinguish an api, strong boundaries make sense, and
+so private by default is better because it forces the writer to at least briefly
+consider where the api boundary should lie. So more noise in files that don’t
+care about an api, or more encouragement to define an api?
+
+I’d suggest private by default within the package as well, e.g. with an annotation
+like `export` to expand vibility.  
+Accidental coupling between internal modules happens all too easily even on
+small teams (even one person communicating with their future self). The argument
+for default private within the package is admittedly not as strong as the argument
+for default private at the package boundary. But on balance
+I think encouraging stronger internal apis is worth the cost of some syntactic noise.
+
+Note that public outside the package could be combined with public inside the package.
+So an element can be public to imports inside the package or public to imports both
+both inside and outside.
+
+### Visibility in Other Lanaguages
+
+* JavaScript and TypeScript are private by default, annotate for public.
```

Comment:
not completely, all members are public by default, opposite to rust. there's the _ prefix convention for private members, and the # standard prefix but for some reason, I have yet to see it in real code.

Python is similar, standard is double underscore but most ppl use a single one.

Reply:
I was thinking of import export visibility in modules. (not object member visibility) 

https://www.typescriptlang.org/docs/handbook/2/modules.html

will make hthat more clear in the text